wsd/DocumentBroker.cpp

```diff
@@ -1491,9 +1491,33 @@ void DocumentBroker::handleUploadToStorageResponse(const StorageBase::UploadResu
             Poco::URI::encode(filename, "", encodedName);
             const std::string filenameAnonym = LOOLWSD::anonymizeUrl(filename);
 
+            constexpr const char* needle = "?access_token";
+            std::size_t index = getPublicUri().toString().find(needle);
+            std::string url1, url2;
+            if (index != std::string::npos)
+            {
+
+                url1 = getPublicUri().toString().substr(0, index);
+                url2 = url.substr(0, index);
```

Comment:
What is `url` and why are we mixing `getPublicUri()` with `url` ?

What happens when `index` is shorter or longer than `url`? 

If they are identical, then let's use the same string please.